integration-test/1491-china-shields.py

```diff
@@ -0,0 +1,244 @@
+            dsl.relation(5, {
+                'name': u'\u5408\u80a5\u7ed5\u57ce\u9ad8\u901f',
+                'name:en': 'G4001 Hefei Ring Expressway',
+                'network': 'CN-expressways', 'ref': 'G4001', 'route': 'road',
+                'source': 'openstreetmap.org', 'type': 'route',
+            }, ways=[65176048]),
+        )
+
+        self.assert_has_feature(
+            z, x, y, 'roads', {
+                'id': 65176048,
+                'shield_text': 'G3', 'network': 'CN:expressways',
+                'all_networks': [
+                    'CN:expressways',
+                    'CN:expressways',
+                    'CN:expressways',
+                    'CN:expressways',
+                    'AsianHighway',
+                ],
+                'all_shield_texts': [
+                    'G3', 'G40', 'G42', 'G4001', 'AH5',
+                ],
+            })
+
+    # China appears to have a fairly common 3rd type of road, signified with an
+    # X000 number. based on the positions of X & S in the ref, it seems like a
+    # less important route type. the network is occasionally given as
+    # "JX-roads"
+    def test_cn_x102(self):
+        import dsl
+
+        z, x, y = (16, 51492, 28421)
+
+        self.generate_fixtures(
+            dsl.is_in('CN', z, x, y),
+            # https://www.openstreetmap.org/way/50677938
+            dsl.way(50677938, dsl.tile_diagonal(z, x, y), {
+                'source': 'openstreetmap.org', 'ref': 'X102;S214',
+                'highway': 'primary',
+            }),
+            dsl.relation(1, {
+                'type': 'route', 'route': 'road', 'ref': 'S214',
+                'name': 'S214', 'source': 'openstreetmap.org',
+            }, ways=[50677938]),
+            dsl.relation(2, {
+                'type': 'route', 'route': 'road', 'ref': 'X102',
+                'name': 'X102', 'source': 'openstreetmap.org',
+            }, ways=[50677938]),
+        )
+
+        self.assert_has_feature(
+            z, x, y, 'roads', {
+                'id': 50677938,
+                'shield_text': 'S214', 'network': 'CN:expressways:regional',
+                'all_shield_texts': ['S214', 'X102'],
+                'all_networks': [
+                    'CN:expressways:regional',
+                    'CN:jx-roads',
+                ],
+            })
+
+    def test_cn_jx_roads_without_relation(self):
+        import dsl
+
+        z, x, y = (16, 51525, 27998)
+
+        self.generate_fixtures(
+            dsl.is_in('CN', z, x, y),
+            # https://www.openstreetmap.org/way/137979647
+            dsl.way(137979647, dsl.tile_diagonal(z, x, y), {
+                'source': 'openstreetmap.org', 'ref': 'X032;X059',
+                'highway': 'secondary',
+            }),
+        )
+
+        self.assert_has_feature(
+            z, x, y, 'roads', {
+                'id': 137979647,
+                'shield_text': 'X032',
+                'network': 'CN:jx-roads',
```

Comment:
What's the provenance of the `CN:jx-roads` designation?

Reply:
This comes from OSM tagging, see next comment [down](https://github.com/tilezen/vector-datasource/pull/1507#discussion_r191597625).